base/database/batch.go

```diff
@@ -0,0 +1,233 @@
+package database
+
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// bulkNow holds a global now state and will be used for each records as
+// CreatedAt and UpdatedAt value if they're empty. This value will be set to the
+// value from gorm.NowFunc() in scopeFromObjects to ensure all objects get the
+// same value.
+// nolint: gochecknoglobals
+var bulkNow time.Time
+
+func BulkInsert(db *gorm.DB, objects []interface{}) error {
+	return bulkExec(db, objects)
+}
+
+// BulkInsertChunk will split the objects passed into the passed chunk size. A
+// slice of errors will be returned (if any).
+func BulkInsertChunk(db *gorm.DB, objects []interface{}, chunkSize int) []error {
+	var allErrors []error
+
+	for {
+		var chunkObjects []interface{}
+
+		if len(objects) <= chunkSize {
+			chunkObjects = objects
+			objects = []interface{}{}
+		} else {
+			chunkObjects = objects[:chunkSize]
+			objects = objects[chunkSize:]
+		}
+
+		if err := bulkExec(db, chunkObjects); err != nil {
+			allErrors = append(allErrors, err)
+		}
+
+		// Nothing more to do
+		if len(objects) < 1 {
+			break
+		}
+	}
+
+	if len(allErrors) > 0 {
+		return allErrors
+	}
+
+	return nil
+}
+
+// bulkExec will convert a slice of interfaces to bulk SQL statement.
+func bulkExec(db *gorm.DB, objects []interface{}) error {
+	scope, err := scopeFromObjects(db, objects)
+	if err != nil {
+		return err
+	}
+
+	// No scope and no error means nothing to do
+	if scope == nil {
+		return nil
+	}
+
+	return db.Exec(scope.SQL, scope.SQLVars...).Error
```

Comment:
This uses `INSERT INTO "test_tables" ("created_at", "deleted_at", "email", "name", "updated_at") VALUES (?, ?, ?, ?, ?), (?, ?, ?, ?, ?), (?, ?, ?, ?, ?) ` in tests. So it adds some extra fields we are not going to use.

Reply:
Behavior is the same as normal `gorm.Create`. The fields were added by `gorm.Model`.